src/main/java/bio/terra/service/filedata/flight/ingest/FileIngestBulkFlight.java

```diff
@@ -136,18 +143,31 @@ public FileIngestBulkFlight(FlightMap inputParameters, Object applicationContext
     addStep(new LoadLockStep(loadService));
     addStep(new IngestFileGetProjectStep(dataset, googleProjectService));
     addStep(new IngestFileGetOrCreateProject(resourceService, dataset), randomBackoffRetry);
-    addStep(new IngestFilePrimaryDataLocationStep(resourceService, dataset), randomBackoffRetry);
-    addStep(new IngestFileMakeBucketLinkStep(datasetBucketDao, dataset), randomBackoffRetry);
+    if (platform.isGcp()) {
+      addStep(new IngestFilePrimaryDataLocationStep(resourceService, dataset), randomBackoffRetry);
+      addStep(new IngestFileMakeBucketLinkStep(datasetBucketDao, dataset), randomBackoffRetry);
+    } else if (platform.isAzure()) {
+      addStep(
+          new IngestFileAzurePrimaryDataLocationStep(resourceService, dataset), randomBackoffRetry);
+      addStep(
+          new IngestFileAzureMakeStorageAccountLinkStep(datasetStorageAccountDao, dataset),
+          randomBackoffRetry);
+    }
 
     if (isArray) {
       addStep(new IngestPopulateFileStateFromArrayStep(loadService));
     } else {
-      addStep(
-          new IngestPopulateFileStateFromFileStep(
-              loadService,
-              appConfig.getMaxBadLoadFileLineErrorsReported(),
-              appConfig.getLoadFilePopulateBatchSize(),
-              gcsPdao));
+      if (platform.isGcp()) {
+        addStep(
+            new IngestPopulateFileStateFromFileStep(
+                loadService,
+                appConfig.getMaxBadLoadFileLineErrorsReported(),
+                appConfig.getLoadFilePopulateBatchSize(),
+                gcsPdao));
+      } else {
+        throw new NotImplementedException(
+            String.format("Ingesting from control file not supported on platform: %s", platform));
```

Comment:
Do we have another ticket to support the arrays scenario with Azure?

Reply:
We do!  DR-1962